### PR TITLE
fix: fix Otelcol installation instructions

### PR DESCRIPTION
**Pull request by astencel-sumo**
## Select the type of change:

- [ ] Minor Changes - Typos, formatting, slight revisions
- [x] Update Content - Revisions and updating sections
- [ ] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, and new packages for the site, Gatsby, React, etc

### CODE REVIEW

File: docs/send-data/opentelemetry-collector/install-collector-linux.md [hunk 1]
```diff
@@ -215,7 +215,7 @@ To run OpenTelemetry Collector as Systemd Service, follow the steps below:
    ```
 1. Get [service file] and save as `/etc/systemd/system/otelcol-sumo.service`:
    ```bash
-   sudo curl https://raw.githubusercontent.com/SumoLogic/sumologic-otel-collector/main/examples/systemd/otelcol-sumo.service -o /etc/systemd/system/otelcol-sumo.service
+   curl https://raw.githubusercontent.com/SumoLogic/sumologic-otel-collector/main/examples/systemd/otelcol-sumo.service | sudo tee /etc/systemd/system/otelcol-sumo.service
```
Comment (sumo-drosiek) [inline docs/send-data/opentelemetry-collector/install-collector-linux.md:218]: Is `tee` available by default on linux distros?

File: docs/send-data/opentelemetry-collector/install-collector-macos.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ You can run the script in two ways:
 
 * By piping `curl` straight into `bash`:
    ```bash
-   sudo curl -s https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh | sudo -E bash -s -- --tag "host.group=default" --tag "deployment.environment=default" && sudo otelcol-sumo --config=/etc/otelcol-sumo/sumologic.yaml --config "glob:/etc/otelcol-sumo/conf.d/*.yaml"
+   curl -Ls https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh | sudo -E bash -s -- --tag "host.group=default" --tag "deployment.environment=default" && sudo otelcol-sumo --config=/etc/otelcol-sumo/sumologic.yaml --config "glob:/etc/otelcol-sumo/conf.d/*.yaml"
```
Comment (dmolenda-sumo) [inline docs/send-data/opentelemetry-collector/install-collector-macos.md:51]: When I run this command, it didn't wait for me to provide a password, and I received the following output:
```
% curl -Ls https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh | sudo -E bash -s -- --tag "host.group=default" --tag "deployment.environment=default" && sudo otelcol-sumo --config=/etc/otelcol-sumo/sumologic.yaml --config "glob:/etc/otelcol-sumo/conf.d/*.yaml"
Password:
zsh: done                   curl -Ls  |
zsh: suspended (tty input)  sudo -E bash -s -- --tag "host.group=default" --tag
```

Comment (astencel-sumo) [reply docs/send-data/opentelemetry-collector/install-collector-macos.md:51]: Hm this is weird. This command "works on my machine" ™️  with zsh 5.9, but gives the above error on Dominika's with Mac's vanilla Zsh 5.8.1. 

I guess for now I will restore the `sudo`s in front of `curl`s in MacOS instructions, we can investigate this independently. For now, the most important thing is to add the missing curl redirection flag `-L` to the MacOS installation command.

Comment (astencel-sumo) [reply docs/send-data/opentelemetry-collector/install-collector-macos.md:51]: Hm this is weird. This command "works on my machine" ™️  with zsh 5.9, but gives the above error on Dominika's with Mac's vanilla Zsh 5.8.1. 

I guess for now I will restore the `sudo`s in front of `curl`s in MacOS instructions, we can investigate this independently. For now, the most important thing is to add the missing curl redirection flag `-L` to the MacOS installation command.

Comment (sumo-drosiek) [reply docs/send-data/opentelemetry-collector/install-collector-macos.md:51]: We should test it on `bash` and not `zsh`

Comment (astencel-sumo) [reply docs/send-data/opentelemetry-collector/install-collector-macos.md]: Restored the `sudo`s.

Comment (astencel-sumo) [reply docs/send-data/opentelemetry-collector/install-collector-macos.md]: > We should test it on `bash` and not `zsh`

The MacOS instructions should probably be tested on the default MacOS's shell.

Comment (sumo-drosiek) [reply docs/send-data/opentelemetry-collector/install-collector-macos.md]: yeeees, I mixed testing instructions vs testing installation script

**Review by sumo-drosiek — APPROVED** [review]

**Review by swiatekm-sumo — APPROVED** [review]

**Review by dmolenda-sumo — APPROVED** [review]

**Pull request merged**